libdevcore/JSON.h

```diff
@@ -23,22 +23,31 @@
 #pragma once
 
 #include <json/json.h>
+#include <sstream>
 
 namespace dev
 {
 
 /// Serialise the JSON object (@a _input) with indentation
 inline std::string jsonPrettyPrint(Json::Value const& _input)
 {
-	return Json::StyledWriter().write(_input);
+	std::stringstream stream;
+	Json::StreamWriterBuilder builder;
+	builder.settings_["indentation"] = "  ";
+	std::unique_ptr<Json::StreamWriter> writer(builder.newStreamWriter());
+	writer->write(_input, &stream);
+	return stream.str();
 }
 
 /// Serialise the JSON object (@a _input) without indentation
 inline std::string jsonCompactPrint(Json::Value const& _input)
 {
-	Json::FastWriter writer;
-	writer.omitEndingLineFeed();
-	return writer.write(_input);
+	std::stringstream stream;
+	Json::StreamWriterBuilder builder;
+	builder.settings_["indentation"] = "";
```

Comment:
The compact layout should not even have newlines. Does it  behave like that?

Reply:
Yes, the behaviour is as expected. To be very sure, I will add a JSON Testsuite that will cover that - `test/libdevcore/JSON.cpp`.